test/runtests.jl

```diff
@@ -116,10 +116,34 @@ using Test
           <summary>Baz</summary>
         </details>
         Qux
-      </details>
-    """
-    @test replace(orig, r"[\s\n]" => "") == replace(HTMLSanitizer.sanitize(orig), r"[\s\n]" => "")
+      </details>"""
+    @test replace(orig, "\n" => "") == replace(HTMLSanitizer.sanitize(orig), "\n" => "")
   end
 end
 
+@testset "preserve relevant whitespace" begin
+  orig = """
+  <!DOCTYPE html>
+  <html>
+    <head>
+      <meta description="test page"></meta>
+    </head>
+    <body>
+      <p>A simple test page.</p>
+      <a></a>
+      <a></a>
+      <pre>
+          <code>
+  foo
+  bar
+  baz
+          </code>
+      </pre>
+    </body>
+  </html>
+  """
+  expected = "<HTML>\n\n  \n    \n  \n  \n    <p>A simple test page.</p>\n    <a></a>\n    <a></a>\n    <pre>\n        <code>\nfoo\nbar\nbaz\n        </code>\n    </pre>\n  \n\n</HTML>"
```

Comment:
Wouldn't it be easier to understand this if it was also triple quoted?

Reply:
Yes, but a bunch of editors remove trailing whitespace and it's annoying to deal with those accidental and almost invisible changes.